lnrpc/walletrpc/walletkit_server.go

```diff
@@ -666,30 +666,61 @@ func (w *WalletKit) SendOutputs(ctx context.Context,
 	// Before we can request this transaction to be created, we'll need to
 	// amp the protos back into the format that the internal wallet will
 	// recognize.
+	var totalOutputValue int64
 	outputsToCreate := make([]*wire.TxOut, 0, len(req.Outputs))
 	for _, output := range req.Outputs {
 		outputsToCreate = append(outputsToCreate, &wire.TxOut{
 			Value:    output.Value,
 			PkScript: output.PkScript,
 		})
+		totalOutputValue += output.Value
 	}
 
 	// Then, we'll extract the minimum number of confirmations that each
 	// output we use to fund the transaction should satisfy.
-	minConfs, err := lnrpc.ExtractMinConfs(req.MinConfs, req.SpendUnconfirmed)
+	minConfs, err := lnrpc.ExtractMinConfs(
+		req.MinConfs, req.SpendUnconfirmed,
+	)
 	if err != nil {
 		return nil, err
 	}
 
+	// Before sending out funds we need to ensure that the remainder of our
+	// wallet funds would cover for the anchor reserve requirement. We'll
+	// also take unconfirmed funds into account.
+	walletBalance, err := w.cfg.Wallet.ConfirmedBalance(
+		0, lnwallet.DefaultAccountName,
```

Comment:
I'd say don't - if unconfirmed funds are considered, it'll probably interfere with our sweeper and leads to undefined behavior.

Reply:
I need to disagree here. Since we only look at the unconfirmed balance for checking if we have enough in reserve, this should be okay. Otherwise you won't be able to create multiple un-confirmed (chained) transactions if you only have a single UTXO using `SendOutputs`.

Reply:
Yes good point re chained transactions. @yyforyongyu I am not familiar with the sweeper yet, what's your concern here?

Reply:
> Otherwise you won't be able to create multiple un-confirmed (chained)

yeah good point.

> what's your concern here?

Our sweeper will sometimes "borrow" utxos from the wallet to construct the sweeping tx, and was concerned they'd compete for the same unconfirmed utxo(also had questions about the coin selection lock). This is not an issue now tho, just checked that the sweeper always use confirmed ones.